test/unit/math/mix/scal/meta/broadcast_array_test.cpp

```diff
@@ -12,4 +13,10 @@ TEST(foo, bar) {
   broadcast_array<fvar<var> > ba(fv);
   EXPECT_EQ(1.0, ba[0].val_.vi_->val_);
   EXPECT_EQ(1.0, ba[2].val_.vi_->val_);
+
+  double two = 2.0;
+  broadcast_array<double> ba2(two);
+  std::vector<double> vd = {{1.0, 2.0}};
+  ba2 = vd;
```

Comment:
I'm not sure I get what's happening here - I see above how a broadcast array will just take the first element... Is that always correct and fine to do? Would anyone be surprised that it dropped the rest of the elements?

Also I would not that this doesn't work with matrices, but I suppose that's probably fine and it just won't compile.

Reply:
Exactly, that is the behaviour I want for my application. People could potentially be surprised... It will only ever be used in my application on vectors of length one. I guess I could have it perform a check to make sure the length of the vector is 1 and throw an error otherwise. Would that not slow things down though? The problem is that the length is dynamic, so it would need to be a runtime check.

Reply:
Yeah, it would slow things down. Do you want to only use the first one because you know that they should be all the same? Or that the first one will correspond to this one in this case?

Reply:
Yep, got that part. But going up a level to operands and partials - is it likely that any time you have a broadcast array here and you try to assign to it, you will also have an array of length one? I'm basically trying to figure out if this is a reasonable API for future users of operands and partials.

Reply:
Yes, I'd say so. You'd have either
- an Eigen vector of length 1;
- a std vector of length 1; or
- a scalar that you'd like to assign to the broadcast array.
What can be problematic, however, is that the Eigen vector may be a composite (e.g. a product of an Eigen matrix and an Eigen vector and is therefore technically an Eigen::Product). This means that it would not suffice to have an assignment statement with r.h.s. Eigen vector. You'd have to include all sorts of complicated Eigen types. Basically, Eigen is a pain when it comes to types. That's why I chose an assignment statement with such a ridiculously generic type. A hack I could imagine could work is to choose an informative name for the assign statement, that would prevent users from misinterpreting its functionality.

Reply:
Github didn't notify me about your reply! I personally think this is fine as long as it acts correctly in the use cases we are expecting. Maybe add a comment about this behavior to the operands and partials high level doc if you don't mind?

Reply:
Great! I'll add the documentation to `broadcast_array`. (Presumably, that's what you meant?) Or would you also like me to add it to `operands_and_partials`? (`operator=` is a method of `broadcast_array`, but `operands_and_partials` uses a `broadcast_array` internally in some instances.

Reply:
I think when you're looking at `broadcast_array` the behavior is reasonably clear (comment drawing attention to it wouldn't be out of place though). But when you're a user of `operands_and_partials`, it would be a really non-obvious piece of behavior when assigning to partials. So I would add the comment to the doc string for `operands_and_partials` somewhere since viewed from that level it's slightly tricky. You can even write that this is almost definitely what they want, but are just drawing attention to it just in case. If you think it's so obviously the correct behavior from the `operands_and_partials` viewpoint that this comment would serve only to distract, then please use your judgment here and ignore this request.

Reply:
Great, I'll add the comments now and will hope that the PR doesn't need to be reapproved before it can be merged.